plugins/woocommerce-blocks/assets/js/editor-components/color-panel/index.tsx

```diff
@@ -142,6 +146,48 @@ export const ColorPanel = ( {
 		attributes,
 		clientId,
 	] );
+
+	const colorContrastWarnings = useMemo( () => {
+		if ( ! settings || settings.length === 0 ) {
+			return;
+		}
+
+		const contrastWarnings = [];
+
+		for ( let i = 0; i < settings.length; i++ ) {
+			const setting = settings[ i ];
+
+			if ( ! setting.colorValue || ! miniCartButtonRef.current ) {
+				continue;
+			}
+
+			const bgColor = getElementBackgroundColor(
+				miniCartButtonRef.current
+			);
+			const message = ContrastChecker( {
+				backgroundColor: bgColor,
+				textColor: setting.colorValue,
+			} );
```

Comment:
@dkotter Calling a component directly (ContrastChecker) is not recommended according to the [React documentation](https://react.dev/reference/rules/react-calls-components-and-hooks#never-call-component-functions-directly), but that is for situations when the component is called inside a JSX return.

In this PR implementation, we need to know beforehand if the component will return null to avoid wrapping it in a div with the label. The only way I figured out to do that is by calling it as a regular function.

Do you see another way to achieve that?

Reply:
Yeah, I think this is fine and don't really see a better approach besides creating our own helper function that basically does the same thing as the `ContractChecker` component, which is not great for maintainability.

Reply:
Thanks for sharing your thoughts!